File: src/main/java/com/github/tuk-cps/ExprParser/ExprTreeLog.java
package ExprParser;

import static jAADD.AADD.REAL;

/**
 * Predefined functions: log
 */
class ExprTreeLog extends ExprTreeFunction {
    ExprTreeLog()  { super(REAL, "log", new ExprTree("x", REAL)); }
    void evalThis() throws ExprError {
        result = getAADDParam(0).log();
    }
}

File: src/main/kotlin/com/github/tuk-cps/exprParser/ExprTreeLog.kt
package exprParser

import jAADD.AADD

/**
 * Predefined functions: log
 */
internal class ExprTreeLog : ExprTreeFunction(AADD.Reals, "log", ExprTree(AADD.Reals, "x")) {
    @Throws(ExprError::class)
    override fun evalUp() {
        value = getAADDParam(0).log()
    }
}